pkg/controller/pxc/upgrade.go

```diff
@@ -364,6 +372,35 @@ func getPodOrderInSts(stsName string, podName string) (int, error) {
 	return strconv.Atoi(podName[len(stsName)+1:])
 }
 
+func (r *ReconcilePerconaXtraDBCluster) waitHostgroups(cr *api.PerconaXtraDBCluster, sfsName string, pod *corev1.Pod, waitLimit int, logger logr.Logger) error {
+	if cr.Spec.HAProxy != nil && cr.Spec.HAProxy.Enabled {
```

Comment:
we are checking only if ProxySQL is enabled.
it is still valid case when HAProxy is disabled and ProxySQL is disabled

Reply:
Resolved in 147448b8efdff4b11a64b862966c2d73f52ef406